.dagger/build/sdk.go

```diff
@@ -135,8 +135,9 @@ func (build *Builder) goSDKContent(ctx context.Context) (*sdkContent, error) {
 	sdkCtrTarball := base.
 		WithEnvVariable("GOTOOLCHAIN", "auto").
 		WithFile("/usr/local/bin/codegen", build.CodegenBinary()).
+		WithExec([]string{"go", "build", "std"}). // pre-cache stdlib
```

Comment:
Yeah I considered that but didn't try yet just due to the plumbing annoyances of getting the go.mod we use by default over here to our CI. Obviously a surmountable problem though.

I'll do something dumb and just hardcode the deps as builds here and see what the impact on image size + initialize time is. If the tradeoff seems worth it I'll figure the plumbing out.

Reply:
Ah jeez, doing this brings initialize time for our CI module down to 57s, but:
1. Adds 5s to the build of the engine itself
   * I'm okay to pay that price since this benefits end users
3. Increases image size from 191MB to 247MB
   * However, a lot of that is the go module download cache, which is something that needs to be downloaded if the engine will use any modules (even non-go ones since their SDKS are based go), so it's if anything more efficient to pack it up here?

I'm airing on the side of doing this, tradeoffs feel worth it right now.

Reply:
Spent a bit going back and forth on this since the tradeoff is a lot closer to call in the case of:
* uncached + ephemeral engine (which requires an image pull)
* single go sdk module with no deps (where the benefit of pre-caching is smaller relative to module with lots of go sdk deps in its dag)

Ephemeral engines are essentially always in some CI system, which typically have good bandwidth. It looks like GHA in [this run](https://github.com/dagger/dagger-for-github/actions/runs/11440058781/job/31825069080) took ~3.9 to pull engine layers on v0.13.5 (165MB), so math'ing it out gives ~5.8s if the engine size goes up to 247MB. +2s.

That's easily worth it for the "lots of go sdk deps" case where we save close to 30s in our CI module, but less clear in the "single go sdk module with no dep" case. However, presumably in those cases the modules are actually doing real work after initialize, in which case the difference of 2s becomes more and more negligible.

Overall, I feel like I'm probably just way overthinking this and will just go with this approach. If users complain about engine image size or notice significant downsides in the ephemeral engine case there's tons of intermediate options we can switch to (i.e. pre-download modules but don't pre-compile them until runtime in the engine, etc.). Won't really know until we try.

Also, ultimately after we get the more baseline fixes in place we're just gonna want to layer on remote module cache anyways which should make many/most of the cases being considered here moot.